repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -135,12 +136,26 @@ class Interpreter(prompt0: Ref[String],
     s"Syntax Error: $msg\n$locationString"
   }
 
-  def processCorrectScript(parsedCode: Parsed.Success[Seq[Seq[String]]],
+  def processCorrectScript(rawParsedCode: Parsed.Success[Seq[( String, Seq[String])]],
                            evaluate: EvaluateCallback): Seq[ImportData] = {
 
+    var offset = 0
+    var parsedCode: Seq[( String, Seq[String])] = Seq()
+    for( (comment, code) <- rawParsedCode.get.value ){
+      val ncomment = comment + "\n"*offset
+      offset = offset + comment.count(_ == '\n') + code.foldLeft(0){ (z,x) =>
+        z + x.count(_ == '\n')
+      } + 1
```

Comment:
What's the `+ 1` at the end for? Let's leave a short comment so people who see this magic number know why it exists and must be `1` (and not `2` or `-1` or some other number)